File: Leetcode/1343. Number of Sub-arrays of Size K and Average Greater than or Equal to Threshold/solution.py
def numOfSubarrays(arr, k, threshold):
  if len(arr ) < k:
    return 0

  countSub = 0
  tempCalc = 0

  for i in range(k):
    tempCalc += arr[i]

    if i == k - 1:
      average = tempCalc/k
      if average >= threshold:
        countSub += 1

  for j in range(k, len(arr)):
    tempCalc = tempCalc - arr[j - k] + arr[j]
    average = tempCalc/k

    if average >= threshold:
      countSub += 1

  return countSub


"print(numOfSubarrays([2,2,2,2,5,5,5,8], 3, 4))"
print(numOfSubarrays([11,13,17,23,29,31,7,5,2,3], 3, 5))

File: Leetcode/1343. Number of Sub-arrays of Size K and Average Greater than or Equal to Threshold/solution.rb
def num_of_subarrays(arr, k, threshold)

  if(arr.size < k)
    return 0
  end

  countSub = 0
  tempCalc = 0

  for i in 0..k - 1
    tempCalc += arr[i]

    if(i == k - 1)
      average = tempCalc/k
      if(average >= threshold)
        countSub += 1
      end
    end

  end

  for j in k..arr.length - 1
    tempCalc = tempCalc - arr[j - k] + arr[j]
    average = tempCalc/k
    if(average >= threshold)
      countSub += 1
    end
  end

  countSub
end

puts num_of_subarrays([2,2,2,2,5,5,5,8], 3, 4)
